dialer_test.go

```diff
@@ -259,21 +276,27 @@ func TestConnectState(main *testing.T) {
 		defer ctrl.Finish()
 
 		l := logger.NewTest()
+		unreadyCh := make(chan error, 10)
 
 		dialer, err := amqpextra.NewDialer(
 			amqpextra.WithURL("amqp://rabbitmq.host"),
-			amqpextra.WithAMQPDial(amqpDialStub(fmt.Errorf("the error"))),
+			amqpextra.WithUnreadyCh(unreadyCh),
 			amqpextra.WithRetryPeriod(time.Millisecond*150),
+			amqpextra.WithAMQPDial(amqpDialStub(fmt.Errorf("the error"))),
 			amqpextra.WithLogger(l),
 		)
 		require.NoError(t, err)
 
 		time.Sleep(time.Millisecond * 100)
-		assertUnready(t, dialer, "the error")
 
+		assertUnready(t, unreadyCh, amqp.ErrClosed.Error())
 		dialer.Close()
 		assertClosed(t, dialer)
 
+		time.Sleep(time.Millisecond * 100)
+
+		assertUnready(t, unreadyCh, "the error")
```

Comment:
do lines 296-298 after `assertUnready(amqp.ErrClose)` and before `dialer.Close()`

Reply:
+